src/main/java/de/photon/anticheataddition/modules/additions/esp/ThirdPersonCameraCanSee.java

```diff
@@ -1,64 +1,83 @@
 package de.photon.anticheataddition.modules.additions.esp;
-
 import de.photon.anticheataddition.util.mathematics.Hitbox;
 import de.photon.anticheataddition.util.mathematics.ResetVector;
 import de.photon.anticheataddition.util.minecraft.world.OcclusionUtil;
 import org.bukkit.Location;
 import org.bukkit.entity.Player;
 import org.bukkit.util.Vector;
 
-final class ThirdPersonCameraCanSee implements CanSee
-{
-    private static final int THIRD_PERSON_OFFSET = 5;
+/**
+ * Checks if a player can see another player in third-person camera view.
+ */
+final class ThirdPersonCameraCanSee implements CanSee {
+
+    /** Maximum distance for third-person camera in Minecraft */
+    private static final int MAX_THIRD_PERSON_DISTANCE = 5;
+
+    /** Offset used to prevent the camera from entering blocks */
     private static final double CAMERA_COLLISION_OFFSET = 0.5D;
 
-    private static double getCameraDistance(final Location eyeLocation, final Vector direction)
-    {
-        final double intersection = OcclusionUtil.getDistanceToFirstIntersectionWithBlock(eyeLocation, direction, THIRD_PERSON_OFFSET);
-        return intersection == 0 ? THIRD_PERSON_OFFSET : intersection - CAMERA_COLLISION_OFFSET;
+    /**
+     * Calculates the actual camera distance considering block intersections.
+     */
+    private static double getCameraDistance(final Location eyeLocation, final Vector direction) {
+        final double intersection = OcclusionUtil.getDistanceToFirstIntersectionWithBlock(
+                eyeLocation, direction, MAX_THIRD_PERSON_DISTANCE);
+
+        return intersection == 0 ? MAX_THIRD_PERSON_DISTANCE : intersection - CAMERA_COLLISION_OFFSET;
     }
 
-    private static Location[] getCameraLocations(final Player player)
-    {
+    /**
+     * Returns all possible camera positions for the player.
+     * Eye level, forward-facing third-person, and backward-facing third-person.
+     */
+    private static Location[] getCameraLocations(final Player player) {
         final Location eyeLocation = player.getEyeLocation();
+        final Vector direction = player.getLocation().getDirection();
 
-        final Vector lookingForward = player.getLocation().getDirection();
-        final Vector lookingBehind = lookingForward.clone().multiply(-1);
-
-        // Do the Cameras intersect with Blocks?
-        // Get the length of the first intersection or 0 if there is none
-        double forwardDistance = getCameraDistance(eyeLocation, lookingForward);
-        double behindDistance = getCameraDistance(eyeLocation, lookingBehind);
+        // Calculate vectors for forward and backward cameras
+        final Vector forwardDir = direction.clone();
+        final Vector backwardDir = direction.clone().multiply(-1);
```

Comment:
worse than before - now it creates 3 Vectors, before it only created two (getDirection creates new vector already)